### PR TITLE
Replace PEP8 by pycodestyle

### LINKED ISSUE #1132: Replace pep8 by pycodestyle in static code analysis
The `pep8` tool has been renamed to `pycodestyle`:
```
pep8 has been renamed to pycodestyle (GitHub issue #466)
Use of the pep8 tool will be removed in a future release.
Please install and use `pycodestyle` instead.
```
We should update our static code analysis scripts.

### LINKED ISSUE #1682: Fix code formatting for all files in the repository once and for all
It seems that not all files in the repository currently conform to our static code style guidelines and people are sometimes hit by this when they edit a file, pertaining files are then checked and unrelated code locations evoke errors.

In order to ease future development, all files in the repository should again be checked for conformity (PEP8, clang-format) and committed.

### PR DESCRIPTION
**Pull request by pnbabu**
Fixes #1132, fixes #1682.

- Replace PEP8 with pycodestyle executable 
- Rename some variables related to Travis CI to Github Actions.
- Remove some unused code related to Travis CI

### CODE REVIEW

File: extras/check_code_style.sh [hunk 1]
```diff
@@ -109,16 +109,16 @@ Options:
                                      Default: --cppcheck=cppcheck
                                      Note: CPPCHECK version 1.69 or later is required.
                                            This corresponds to the version installed in
-                                           the NEST Travis CI build and test environment.
+                                           the NEST Github Actions CI build and test environment.
```
Comment (jougs) [inline extras/check_code_style.sh:112]:
```suggestion
                                           the NEST CI build and test environment.
```

File: extras/check_code_style.sh [hunk 2]
```diff
@@ -109,16 +109,16 @@ Options:
...
 
     --clang-format=exe               The name of the CLANG-FORMAT executable.
                                      Default: --clang-format=clang-format-3.6
                                      Note: CLANG-FORMAT version 3.6 is required.
                                            This corresponds to the version installed in
-                                           the NEST Travis CI build and test environment.
+                                           the NEST Github Actions CI build and test environment.
```
Comment (jougs) [inline extras/check_code_style.sh:118]:
```suggestion
                                           the NEST CI build and test environment.
```

File: extras/check_code_style.sh [hunk 3]
```diff
@@ -76,7 +76,7 @@ print_usage() {
 Usage: ./extras/check_code_style.sh [options ...]
 
 This script processes C/C++ and Python source code files to verify compliance with the NEST
-coding  style  guidelines.  The  checks are performed the same way as in the NEST Travis CI
+coding  style  guidelines.  The  checks are performed the same way as in the NEST Github Actions CI
```
Comment (jougs) [inline extras/check_code_style.sh:79]: Let's make our life easier the next time we change CI systems ;-)

```suggestion
coding  style  guidelines.  The  checks are performed the same way as in the NEST CI
```

File: extras/check_code_style.sh [hunk 4]
```diff
@@ -280,9 +280,9 @@ if $PERFORM_CLANG_FORMAT; then
   fi
 fi
 
-# Verify the PEP8 installation.
+# Verify the pycodestyle installation.
 if $PERFORM_PEP8; then
-  $PEP8 --ignore="E121,E501" ./extras/parse_build_log.py || error_exit "Failed to verify the PEP8 installation. Executable: $PEP8"
+  $PEP8 --ignore="E121,E501,W504" ./extras/parse_build_log.py || error_exit "Failed to verify the pycodestyle installation. Executable: $PEP8"
```
Comment (jougs) [inline extras/check_code_style.sh:285]: I just fixed some of these W504 warnings (line break after binary operator) in a branch of mine and in return got a bunch of W503 (line break *before* binary operator) :disappointed:

As I think the readability of those code locations depends more on context and a general rule will always fail in some cases, I suggest to ignore both: 


```suggestion
  $PEP8 --ignore="E121,E501,W503,W504" ./extras/parse_build_log.py || error_exit "Failed to verify the pycodestyle installation. Executable: $PEP8"
```

File: extras/check_code_style.sh [hunk 5]
```diff
@@ -303,18 +303,18 @@ if [ ! -x ./extras/static_code_analysis.sh ]; then
 fi
 
 
-RUNS_ON_TRAVIS=false
+RUNS_ON_GITHUB_ACTIONS=false
```
Comment (jougs) [inline extras/check_code_style.sh:306]:
```suggestion
RUNS_ON_CI=false
```

File: extras/check_code_style.sh [hunk 6]
```diff
@@ -303,18 +303,18 @@ if [ ! -x ./extras/static_code_analysis.sh ]; then
...
 
 unset NEST_VPATH               # These command line arguments are placeholders and not required here.
 IGNORE_MSG_VERA=false          # They are needed when running the static code analysis script within
-IGNORE_MSG_CPPCHECK=false      # the Travis CI build environment.
+IGNORE_MSG_CPPCHECK=false      # the Github Actions CI build environment.
```
Comment (jougs) [inline extras/check_code_style.sh:310]:
```suggestion
IGNORE_MSG_CPPCHECK=false      # the CI build environment.
```

File: extras/check_code_style.sh [hunk 7]
```diff
@@ -303,18 +303,18 @@ if [ ! -x ./extras/static_code_analysis.sh ]; then
...
 IGNORE_MSG_CLANG_FORMAT=false
-IGNORE_MSG_PEP8=false
+IGNORE_MSG_PYCODESTYLE=false
 
-./extras/static_code_analysis.sh "$RUNS_ON_TRAVIS" "$INCREMENTAL" "$file_names" "$NEST_VPATH" \
+./extras/static_code_analysis.sh "$RUNS_ON_GITHUB_ACTIONS" "$INCREMENTAL" "$file_names" "$NEST_VPATH" \
```
Comment (jougs) [inline extras/check_code_style.sh:314]:
```suggestion
./extras/static_code_analysis.sh "$RUNS_ON_CI" "$INCREMENTAL" "$file_names" "$NEST_VPATH" \
```

File: extras/gha_build.sh [hunk 8]
```diff
@@ -20,8 +20,8 @@
 # along with NEST.  If not, see <http://www.gnu.org/licenses/>.
 
 
-# This shell script is part of the NEST Travis CI build and test environment.
-# It is invoked by the top-level Travis script '.travis.yml'.
+# This shell script is part of the NEST Github Actions CI build and test environment.
```
Comment (jougs) [inline extras/gha_build.sh:23]:
```suggestion
# This shell script is part of the NEST CI build and test environment.
```

File: extras/gha_build.sh [hunk 9]
```diff
@@ -122,21 +112,21 @@ if [ "$xNEST_BUILD_TYPE" = "STATIC_CODE_ANALYSIS" ]; then
     PERFORM_PEP8=true
 
     # The following command line parameters indicate whether static code analysis error messages
-    # will cause the Travis CI build to fail or are ignored.
+    # will cause the Github Actions CI build to fail or are ignored.
```
Comment (jougs) [inline extras/gha_build.sh:115]:
```suggestion
    # will cause the CI build to fail or are ignored.
```

File: extras/gha_build.sh [hunk 10]
```diff
@@ -122,21 +112,21 @@ if [ "$xNEST_BUILD_TYPE" = "STATIC_CODE_ANALYSIS" ]; then
...
     IGNORE_MSG_VERA=false
     IGNORE_MSG_CPPCHECK=true
     IGNORE_MSG_CLANG_FORMAT=false
-    IGNORE_MSG_PEP8=false
+    IGNORE_MSG_PYCODESTYLE=false
 
-    # The script is called within the Travis CI environment and thus can not be run incremental.
-    #RUNS_ON_TRAVIS=true
+    # The script is called within the Github Actions CI environment and thus can not be run incremental.
+    RUNS_ON_GITHUB_ACTIONS=true
```
Comment (jougs) [inline extras/gha_build.sh:122]:
```suggestion
    # The script is called within the CI environment and thus can not be run incremental.
    RUNS_ON_CI=true
```

File: extras/gha_build.sh [hunk 11]
```diff
@@ -122,21 +112,21 @@ if [ "$xNEST_BUILD_TYPE" = "STATIC_CODE_ANALYSIS" ]; then
...
     INCREMENTAL=false
 
     chmod +x extras/static_code_analysis.sh
-    ./extras/static_code_analysis.sh "$RUNS_ON_TRAVIS" "$INCREMENTAL" "$file_names" "$NEST_VPATH" \
+    ./extras/static_code_analysis.sh "$RUNS_ON_GITHUB_ACTIONS" "$INCREMENTAL" "$file_names" "$NEST_VPATH" \
```
Comment (jougs) [inline extras/gha_build.sh:126]:
```suggestion
    ./extras/static_code_analysis.sh "$RUNS_ON_CI" "$INCREMENTAL" "$file_names" "$NEST_VPATH" \
```

File: extras/static_code_analysis.sh [hunk 12]
```diff
@@ -20,8 +20,8 @@
 # along with NEST.  If not, see <http://www.gnu.org/licenses/>.
 
 
-# This shell script is part of the NEST Travis CI build and test environment.
-# It performs the static code analysis and is invoked by 'travis_build.sh'.
+# This shell script is part of the NEST Github Actions CI build and test environment.
```
Comment (jougs) [inline extras/static_code_analysis.sh:23]:
```suggestion
# This shell script is part of the NEST CI build and test environment.
```

File: extras/static_code_analysis.sh [hunk 13]
```diff
@@ -20,8 +20,8 @@
...
+# It performs the static code analysis and is invoked by 'gha_build.sh'.
```
Comment (jougs) [inline extras/static_code_analysis.sh:24]: I suggest to rename `extras/gha_build.sh` to `ci_build.sh` and change all references to it accordingly.

```suggestion
# It performs the static code analysis and is invoked by 'ci_build.sh'.
```

File: extras/static_code_analysis.sh [hunk 14]
```diff
@@ -32,8 +32,8 @@
 #
 
 # Command line parameters.
-RUNS_ON_TRAVIS=${1}           # true or false, indicating whether the script is executed on Travis CI or runs local.
-INCREMENTAL=${2}              # true or false, user needs to confirm befor checking a source file.
+RUNS_ON_GITHUB_ACTIONS=${1}   # true or false, indicating whether the script is executed on Github Actions CI or runs local.
```
Comment (jougs) [inline extras/static_code_analysis.sh:35]:
```suggestion
RUNS_ON_CI=${1}               # true or false, indicating whether the script is executed on the CI or not.
```

File: extras/static_code_analysis.sh [hunk 15]
```diff
@@ -47,15 +47,15 @@ PERFORM_PEP8=${12}            # true or false, indicating whether PEP8 analysis
...
 IGNORE_MSG_CLANG_FORMAT=${15} # true or false, indicating whether CLANG-FORMAT messages should accout for the build result.
-IGNORE_MSG_PEP8=${16}         # true or false, indicating whether PEP8 messages should accout for the build result.
+IGNORE_MSG_PYCODESTYLE=${16}  # true or false, indicating whether pycodestyle messages should accout for the build result.
 
-# PEP8 rules to ignore.
-PEP8_IGNORES="E121,E123,E126,E226,E24,E704"
-PEP8_IGNORES_EXAMPLES="${PEP8_IGNORES},E402"
-PEP8_IGNORES_TOPO_MANUAL="${PEP8_IGNORES_EXAMPLES},E265"
+# PYCODESTYLE rules to ignore.
+PYCODESTYLE_IGNORES="E121,E123,E126,E226,E24,E704"
```
Comment (jougs) [inline extras/static_code_analysis.sh:53]: Please also add W503 and W504 here. See [above](https://github.com/nest/nest-simulator/pull/2105/files#diff-cb3c61b3c773983f77fd537c8923e562835078815e095a308a10d176cfe54a1bR285) for why.

I also dislike it a bit that the verification above uses its own set of ignores and does not use the list defined here. How complicated would it be to do so?

File: extras/static_code_analysis.sh [hunk 16]
```diff
@@ -65,10 +65,10 @@ ROOT_DIRS_TO_IGNORE="thirdparty"
 DIRS_TO_IGNORE=$(for dir in ${ROOT_DIRS_TO_IGNORE}; do find ${dir} -type d; done)
 
 # Print a message.
-# The format of the message depends on whether the script is executed on Travis CI or runs local.
+# The format of the message depends on whether the script is executed on Github Actions CI or runs local.
```
Comment (jougs) [inline extras/static_code_analysis.sh:68]:
```suggestion
# The format of the message depends on whether the script is executed on CI or not.
```

File: extras/static_code_analysis.sh [hunk 17]
```diff
@@ -65,10 +65,10 @@ ROOT_DIRS_TO_IGNORE="thirdparty"
...
 # print_msg "string1" "string2"
 print_msg() {
-  if $RUNS_ON_TRAVIS; then
+  if $RUNS_ON_GITHUB_ACTIONS; then
```
Comment (jougs) [inline extras/static_code_analysis.sh:71]:
```suggestion
  if $RUNS_ON_CI; then
```

File: extras/static_code_analysis.sh [hunk 18]
```diff
@@ -91,14 +91,14 @@ if $PERFORM_CLANG_FORMAT; then
...
+  PYCODESTYLE_VERS=`$PEP8 --version`
+  print_msg "MSGBLD0105: " "PEP8         : $PYCODESTYLE_VERS"
 fi
 print_msg "" ""
 
 # The following messages report on the command line arguments IGNORE_MSG_xxx which indicate whether
-# static code analysis error messages will cause the Travis CI build to fail or are ignored.
-if $RUNS_ON_TRAVIS; then
+# static code analysis error messages will cause the GithubActions CI build to fail or are ignored.
+if $RUNS_ON_GITHUB_ACTIONS; then
```
Comment (jougs) [inline extras/static_code_analysis.sh:101]:
```suggestion
# static code analysis error messages will cause the CI build to fail or are ignored.
if $RUNS_ON_CI; then
```

File: extras/static_code_analysis.sh [hunk 19]
```diff
@@ -108,8 +108,8 @@ if $RUNS_ON_TRAVIS; then
   if $IGNORE_MSG_CLANG_FORMAT; then
     print_msg "MSGBLD1030: " "IGNORE_MSG_CLANG_FORMAT is set. CLANG_FORMAT messages will not cause the build to fail."
   fi
-  if $RUNS_ON_TRAVIS && $IGNORE_MSG_PEP8; then
-    print_msg "MSGBLD1040: " "IGNORE_MSG_PEP8 is set. PEP8 messages will not cause the build to fail."
+  if $RUNS_ON_GITHUB_ACTIONS && $IGNORE_MSG_PYCODESTYLE; then
```
Comment (jougs) [inline extras/static_code_analysis.sh:111]:
```suggestion
  if $RUNS_ON_CI && $IGNORE_MSG_PYCODESTYLE; then
```

File: extras/static_code_analysis.sh [hunk 20]
```diff
@@ -137,12 +137,12 @@ for f in $FILE_NAMES; do
     print_msg "MSGBLD0110: " "$f is not a file or does not exist anymore."
     continue
   fi
-  if ! $RUNS_ON_TRAVIS && $INCREMENTAL; then
+  if ! $RUNS_ON_GITHUB_ACTIONS && $INCREMENTAL; then
```
Comment (jougs) [inline extras/static_code_analysis.sh:140]:
```suggestion
  if ! $RUNS_ON_CI && $INCREMENTAL; then
```

File: extras/static_code_analysis.sh [hunk 21]
```diff
@@ -137,12 +137,12 @@ for f in $FILE_NAMES; do
...
     print_msg "" ""
     print_msg "" "Press [Enter] to continue.  (Static code analysis for file $f.)"
     read continue
   fi
-  if $RUNS_ON_TRAVIS; then
+  if $RUNS_ON_GITHUB_ACTIONS; then
```
Comment (jougs) [inline extras/static_code_analysis.sh:145]:
```suggestion
  if $RUNS_ON_CI; then
```

File: extras/static_code_analysis.sh [hunk 22]
```diff
@@ -152,7 +152,7 @@ for f in $FILE_NAMES; do
       cppcheck_failed=false
       clang_format_failed=false
 
-      if $RUNS_ON_TRAVIS; then
+      if $RUNS_ON_GITHUB_ACTIONS; then
```
Comment (jougs) [inline extras/static_code_analysis.sh:155]:
```suggestion
      if $RUNS_ON_CI; then
```

File: extras/static_code_analysis.sh [hunk 23]
```diff
@@ -170,7 +170,7 @@ for f in $FILE_NAMES; do
           done
         fi
         rm ${f_base}_vera.txt
-        if $RUNS_ON_TRAVIS; then
+        if $RUNS_ON_GITHUB_ACTIONS; then
```
Comment (jougs) [inline extras/static_code_analysis.sh:173]:
```suggestion
        if $RUNS_ON_CI; then
```

File: extras/static_code_analysis.sh [hunk 24]
```diff
@@ -187,7 +187,7 @@ for f in $FILE_NAMES; do
           cat ${f_base}_cppcheck.txt | while read line
           do
             print_msg "MSGBLD0155: " "[CPPC] $line"
-            if $RUNS_ON_TRAVIS; then
+            if $RUNS_ON_GITHUB_ACTIONS; then
```
Comment (jougs) [inline extras/static_code_analysis.sh:190]:
```suggestion
            if $RUNS_ON_CI; then
```

File: extras/static_code_analysis.sh [hunk 25]
```diff
@@ -197,7 +197,7 @@ for f in $FILE_NAMES; do
           done
         fi
         rm ${f_base}_cppcheck.txt
-        if $RUNS_ON_TRAVIS; then
+        if $RUNS_ON_GITHUB_ACTIONS; then
```
Comment (jougs) [inline extras/static_code_analysis.sh:200]:
```suggestion
        if $RUNS_ON_CI; then
```

File: extras/static_code_analysis.sh [hunk 26]
```diff
@@ -219,7 +219,7 @@ for f in $FILE_NAMES; do
         fi
         rm $file_formatted
         rm $file_diff
-        if $RUNS_ON_TRAVIS; then
+        if $RUNS_ON_GITHUB_ACTIONS; then
```
Comment (jougs) [inline extras/static_code_analysis.sh:222]:
```suggestion
        if $RUNS_ON_CI; then
```

File: extras/static_code_analysis.sh [hunk 27]
```diff
@@ -231,29 +231,29 @@ for f in $FILE_NAMES; do
...
+        if ! pycodestyle_result=`$PEP8 --max-line-length=$PYCODESTYLE_MAX_LINE_LENGTH --ignore=$IGNORES $f` ; then
+          printf '%s\n' "$pycodestyle_result" | while IFS= read -r line
           do
             print_msg "MSGBLD0195: " "[PEP8] $line"
           done
           # Add the file to the list of files with format errors.
           python_files_with_errors="$python_files_with_errors $f"
         fi
-        if $RUNS_ON_TRAVIS; then
+        if $RUNS_ON_GITHUB_ACTIONS; then
```
Comment (jougs) [inline extras/static_code_analysis.sh:256]:
```suggestion
        if $RUNS_ON_CI; then
```

File: extras/static_code_analysis.sh [hunk 28]
```diff
@@ -282,7 +282,7 @@ if [ $nlines_copyright_check \> 1 ] || \
     for f in $c_files_with_errors; do
       print_msg "MSGBLD0220: " "... $f"
     done
-    if ! $RUNS_ON_TRAVIS; then
+    if ! $RUNS_ON_GITHUB_ACTIONS; then
```
Comment (jougs) [inline extras/static_code_analysis.sh:285]:
```suggestion
    if ! $RUNS_ON_CI; then
```

File: extras/static_code_analysis.sh [hunk 29]
```diff
@@ -311,13 +311,13 @@ if [ $nlines_copyright_check \> 1 ] || \
     for f in $python_files_with_errors; do
       print_msg "MSGBLD0220: " "... $f"
     done
-    if ! $RUNS_ON_TRAVIS; then
+    if ! $RUNS_ON_GITHUB_ACTIONS; then
```
Comment (jougs) [inline extras/static_code_analysis.sh:314]:
```suggestion
    if ! $RUNS_ON_CI; then
```

File: extras/static_code_analysis.sh [hunk 30]
```diff
@@ -311,13 +311,13 @@ if [ $nlines_copyright_check \> 1 ] || \
...
       print_msg "" "Run pep8ify -w <filename> for autocorrection."
       print_msg "" ""
     fi
   fi
 
-  if ! $RUNS_ON_TRAVIS; then
+  if ! $RUNS_ON_GITHUB_ACTIONS; then
```
Comment (jougs) [inline extras/static_code_analysis.sh:320]:
```suggestion
  if ! $RUNS_ON_CI; then
```

**Review by jougs — CHANGES_REQUESTED** [review]
Many thanks for your contribution. I have added a couple of comments and suggestions, most of which you can probably just batch commit.

File: extras/static_code_analysis.sh (hunk 15 above)
Comment (pnbabu) [reply extras/static_code_analysis.sh:53]: You are right. I will make a list of error and warning codes and pass them to this script.

File: extras/static_code_analysis.sh (hunk 13 above)
Comment (pnbabu) [reply extras/static_code_analysis.sh]: Done

Comment (jougs) [reply extras/static_code_analysis.sh]: Now, the script is called just `ci.sh`, which I find a bit too brief. Can you please make it `ci_build.sh`? Or is there a reason for your shorter choice?

Comment (pnbabu) [reply extras/static_code_analysis.sh]: I somehow read your comment wrongly and assumed it as `ci.sh`. Sorry, my bad. I have fixed this correctly now as `ci_build.sh`.

**Comment by jougs** [comment]
@pnbabu: I have sent you https://github.com/pnbabu/nest-simulator/pull/1, as I thought this would fit nicely here. If you decide to merge it, please also make to mark #1682 as being fixed by this PR.

**Comment by jougs** [comment]
Fixes #1682.

**Review by jougs — APPROVED** [review]
LGTM! Merging without second review, once the CI is happy.

**Pull request merged**